pyomo/common/tests/test_config.py

```diff
@@ -1479,12 +1479,8 @@ def test_block_keys(self):
         keys = self.config['scenario'].keys()
         # lists are independent
         self.assertFalse(keys is self.config['scenario'].keys())
-        if PY3:
-            self.assertIsNot(type(keys), list)
-            self.assertEqual(list(keys), ref)
-        else:
-            self.assertIs(type(keys), list)
-            self.assertEqual(keys, ref)
+        self.assertIs(type(keys), list)
+        self.assertEqual(list(keys), ref)
```

Comment:
You kept the wrong (python 2) branch of the IF here.

Reply:
Ah, maybe you can help shed some light on this. The tests fail if we run:
```
self.assertIsNot(type(keys/values/items), list)
```
They return this error (same error for all three checks on type):
```
======================================================================
FAIL: test_block_values (pyomo.common.tests.test_config.TestConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyomo/pyomo/pyomo/common/tests/test_config.py", line 1504, in test_block_values
    self.assertIsNot(type(values), list)
AssertionError: unexpectedly identical: <class 'list'>
```

Ergo, `type(values/keys/items)` is returning as a list.

Reply:
That was testing that the `ConfigDict` class behaved like `dict`: in Python 2, `keys`, `values`, and `items` return a `list` and `iterkeys`, ` itervalues`, and `iteritems` return generator-like things.  In Python 3, they made `keys`, `values`, and `items` return the generator-like things and got rid of the `iter`* methods.  The way we mocked that up here was to define things using Python 2 style, and then if we were running Python 3, we overrode the `keys`, `values`, and `items` with their `iter`* implementations.  You removed the magic that redefined `keys`, `values`, and `items`  (old lines 1905-1910), but still need to explicitly change the API for Python 3.

This was what I was getting at in my comment above for old lines 1905-1910.